rotkehlchen/db/upgrades/v31_v32.py

```diff
@@ -86,6 +86,23 @@ def _add_new_tables(cursor: 'Cursor') -> None:
 );""")  # noqa: E501
 
 
+def _update_history_entries_from_kraken(cursor: 'Cursor') -> None:
+    """The logic for kraken was adding additional entries for trades when fee + kfee was
+    being used. This function makes the state of the database consistent with the upgraded
+    logic by:
+    - Removing extra row additions
+    - Updating the sequence index to avoid duplicates
+    """
+    cursor.execute("""
+    DELETE FROM history_events where location="B" AND asset="KFEE" AND
+     type="trade" AND subtype=NULL;
+    """)
+    cursor.execute("""
```

Comment:
> because the check can't be done in sqlite as far as I know 

It can. How do you think I found it out in the first place? I posted it yesterday in discord:

```
SELECT e.event_identifier, e.sequence_index from history_events e JOIN (SELECT event_identifier, sequence_index, COUNT(*) as cnt FROM history_events GROUP BY event_identifier, sequence_index) other ON e.event_identifier = other.event_identifier and e.sequence_index=other.sequence_index WHERE other.cnt > 1;
```

The main question would be what to do if say we (1) run your upgrade change and (2) run this check and the check still gives a result.

If it still gives a result we would have to decide what to do with it. At DB upgrade time we probably would not be able to know. Only "solution" would be an ugly one and it would mean to just assign a random sequence index to any such entries.

Perhaps I am overthinking it. Not sure. I just don't want to have people stuck at failed upgrades. Keep in mind we still don't have the user editing tool for the DB.

Reply:
I was thinking in handling the update and everything from sqlite but you are right, probably can be done. Yes, having people stuck with upgrades would be the real issue